front/src/usermanagement/UserManagementView.js

```diff
@@ -481,15 +575,16 @@ class UserManagementView extends Component {
 
   update() {
     const tempRows = [];
-    this.state.userList.forEach((user) => {
-      if (this.state.username !== user.name) {
-        const row = this.createData(user.name,
-          user.role,
-          this.returnPassButton(user.id, manage, fin),
-          this.returnRemoveButton(user.id, manage, fin));
+    for (const i in this.state.userList) {
+      if (localStorage.taseraUserName !== this.state.userList[i].name) {
+        const row = this.createData(this.state.userList[i].name,
+          this.state.userList[i].role,
+          this.returnPassButton(this.state.userList[i].id, manage, fin),
+          this.returnRemoveButton(this.state.userList[i].id, manage, fin),
+          this.returnAddEmailButton(this.state.userList[i].id, manage, fin));
```

Comment:
Is there a possibility of not incorporating the state as props and instead reference the state by itself inside the button? May not be possible but if it is do that